rssfeed/tests/test_utils.py

```diff
@@ -0,0 +1,171 @@
+        feed_mock = Mock(spec=Feed)
+        feed_mock.xml_url = "test-feed-url"
+        feed_mock.published_time = None
+        feed_mock.image = \
+            "http://news.bbcimg.co.uk/nol/shared/img/bbc_news_120x60.gif"
+        feed_mock.description = "BBC News - Home"
+        self.feed_mock = feed_mock
+
+    def test_published_time(self):
+        """Test Published Time variations"""
+        with patch("rssfeed.utils.feedparser.parse", self.parser_mock):
+            # No published time in DB
+            feed_mock = self.feed_mock
+            feed_mock.published_time = None
+            poll_feed(feed_mock, verbose=True)
+
+        # Published time in DB later than on feed
+        feed_mock.published_time = pytz.utc.localize(
+            datetime(2017, 01, 01, 13, 0, 0)
+        )
+        poll_feed(feed_mock, verbose=True)
+
+    def test_missing_attribute(self):
+        """Test with missing attribute: description_detail"""
+        parser_mock = self.parser_mock
+        del parser_mock.return_value.feed.description_detail
+        with patch("rssfeed.utils.feedparser.parse", parser_mock):
+            poll_feed(self.feed_mock, verbose=True)
+
+    def test_with_feed_description(self):
+        """Test with description_detail present"""
+        parser_mock = self.parser_mock
+        parser_mock.return_value.feed.description_detail = "text/plain"
+        parser_mock.return_value.feed.description = "BBC News - Home"
+        with patch("rssfeed.utils.feedparser.parse", parser_mock):
+            poll_feed(self.feed_mock, verbose=True)
+
+    def test_missing_image_attribute(self):
+        """Test with missing image attibute"""
+        parser_mock = self.parser_mock
+        parser_mock.return_value.feed.description_detail = "text/plain"
+        parser_mock.return_value.feed.description = "BBC News - Home"
+        del parser_mock.return_value.feed.image
+        with patch("rssfeed.utils.feedparser.parse", parser_mock):
+            poll_feed(self.feed_mock, verbose=True)
+
+
+@patch("rssfeed.utils.feedparser.parse")
+class PollFeedBozoExceptionTest(TestCase):
+    """
+    Test polling feeds where Bozo Exception returned.
+    i.e. The exception raised when attempting to parse a non-well-formed feed.
+    """
+
+    def setUp(self):
+        feed_mock = Mock(spec=Feed)
+        feed_mock.xml_url = "test-feed-url"
+        feed_mock.published_time = None
+        self.feed_mock = feed_mock
+
+    def test_bozo_exception(self, parse_mock):
+        """Test with Bozo Exception returned"""
+        parse_mock.return_value.feed.bozo_exception = "bozo_exception returned"
+        poll_feed(self.feed_mock, verbose=True)
+
+
+class PollEntriesTest(TestCase):
+    def setUp(self):
+        """Create feedparser.parse_mock object"""
+        parser_mock = Mock()
+        del parser_mock.return_value.feed.bozo_exception
+        parser_mock.return_value.feed.published_parsed = (
+            2017, 01, 01,
```

Comment:
Be **very** careful when padding numbers with zeros. It will be interpreted as _octal_ numbers. This happens to work, but something like `09` would cause an error.